### ISSUE TITLE
Accessibility: Should elements not defined in the config be in the DOM tree of the header?

### ISSUE THREAD

**Issue by cosbyr** [Esri/global-nav#300]
- Right now it appears the default behavior of the global-nav is to still create these elements as hidden.

Example:
In the implementation below... the search, inline search, shopping cart, notifications, and app launcher parts of the global-nav are not defined in the config as desired, yet they show in the DOM tree:
![2020-06-15_9-46-11](https://user-images.githubusercontent.com/11748268/84684155-1d489680-aeed-11ea-8579-615088b05c7a.gif)
![image](https://user-images.githubusercontent.com/11748268/84683783-985d7d00-aeec-11ea-95a1-54cfaf89901f.png)

See similar issue regarding the inline title: #297

This causes issues for Accessibility/Screen Readers:
![image](https://user-images.githubusercontent.com/11748268/84684969-4c133c80-aeee-11ea-8544-729a0aef7178.png)

**Comment by paulcpederson** [comment]
@cosbyr so we can't omit them from the DOM due to peculiarities with how this library was written. Essentially all of these components are set up and added to the DOM empty and they listen to events and populate themselves later. I've added a change where they are at least hidden (`display: none`). This will hide them from screen readers and fix some layout issues. 

#314

**Comment by paulcpederson** [comment]
Should be fixed in [📦  v1.4.0](https://github.com/Esri/global-nav/releases/tag/v1.4.0)

**Comment by MikeTschudi** [comment]
@paulcpederson, they're still being created:
![image](https://user-images.githubusercontent.com/2125181/90449453-55f71e80-e09c-11ea-8436-8cad08af6d61.png)

**Comment by paulcpederson** [comment]
@MikeTschudi Yes, see above comment. We cannot remove them from the dom. I have marked as closed because they now have `display: none` so shouldn't be picked up by screen readers or effect layout.

**Comment by MikeTschudi** [comment]
The hidden `<div class="esri-header-search">` element gets focus as one tabs.

Why can't these unused hidden items be removed from the dom, please? We're removing them as a patch--is this a mistake?
```typescript
this.removeElement(document.getElementsByClassName("esri-header-apps")[0]);
this.removeElement(document.getElementsByClassName("esri-header-inlineSearch")[0]);
this.removeElement(document.getElementsByClassName("esri-header-notifications")[0]);
this.removeElement(document.getElementsByClassName("esri-header-search")[0]);
this.removeElement(document.getElementsByClassName("esri-header-shopping-cart")[0]);
```

**Comment by paulcpederson** [comment]
It's not a mistake, it's a byproduct of how the library was built. Here's an overview of the control flow (which took me a long time to understand...):

1. When you call `create()`, first, all sections add DOM to the header with no content from config populated
2. These sections add an event listener to their dom node for a custom event about header updates
3. the create method then fires the event with your initial config
4. the section will update itself with your actual content
5. (optional) dev can push additional updates either directly on the dom node or on the entire header by dispatching events on either element

So the problem is this: currently, if you don't create this dom initially, nothing will render (even for those who have defined it in the config). If we don't put the dom in, you will also be unable to make granular updates later. If we remove the dom when it's not defined in the config, the dev will be unable to find said dom and dispatch events on it later. The more interactive sections all work with this way.

I wish deep down in my heart it was as simple as `if (apps) { // add dom }`, but for <strike>better</strike> or worse it's much more complicated. My fix was to leave in dom but hide to see if that helped folks, but based on the other issue about this same topic, it didn't. I will keep thinking though ways around this.

**Comment by MikeTschudi** [comment]
@paulcpederson, thank you for the detailed explanation. What I meant by mistake was my removing the elements. Perhaps removing the elements after global nav has initialized (which is what I do) is OK because no further events are likely for those empty items?

**Comment by paulcpederson** [comment]
I think you should be ok if you won't need those elements. Once it's gone, it's gone for good, though. So you won't be able to say, initialize without an app switcher and then add one later unless you re-create the whole thing again.

**Comment by MikeTschudi** [comment]
Makes sense; thank you
